bin/cylc-check-software

```diff
@@ -234,9 +249,41 @@ def functionality_print(func):
     return
 
 
+def individual_status_print(module):
+    """Return a pass (0) or fail (1) result for an individual module check."""
+    if module == 'Python':
+        return int(not check_py_ver(*req_py_ver_range, write=False))
+    if module in opt_spec.keys():
+        ver_req, _, tag = opt_spec[module][:3]
+        if tag == 'PY':
+            return int(not check_py_module_ver(module, ver_req, write=False))
+        elif tag == 'TEX':
+            return int(not tex_module_search(module, write=False))
+        elif tag == 'OTHER':
+            other_args = opt_spec[module][3]
+            return int(
+                not cmd_find_ver(module, ver_req, *other_args, write=False))
+
+
 def main():
-    """Test for and print external software packages with minimum versions
-       as required for both minimal core and fully-functional cylc."""
+    """Test for Cylc software dependency/dependencies and print result(s).
+
+    Check for external software packages with minimum versions as required
+    for both minimal core and fully-functional Cylc. If an argument is suppied,
+    test for that module only and return '0' for a pass and '1' for a fail,
+    else check for all package dependencies and print the results in a table.
+    """
+    # Check for valid module argument; if present return 0 or 1 for pass/fail.
+    if len(sys.argv) > 1:
```

Comment:
Could we check for multiple dependencies with a for loop here? Also, I think it would be nicer to signal success/failure through exit code rather than STDOUT. How about the following?

```python
    exit_code = 0
    for cmd_arg in sys.argv[1:]:
        mod_arg = upper_case_conv.get(cmd_arg.lower(), cmd_arg.lower())
        if mod_arg not in module_args:
            sys.stderr.write('No such module in the software dependencies: ' +
                             cmd_arg + '\n')
            exit_code = max(exit_code, 2)
        else:
            exit_code = max(exit_code, individual_status_print(mod_arg))
    if len(sys.argv) > 1:
        sys.exit(exit_code)
```

Reply:
Yes, sorry, an exit code would be more appropriate. Okay, we can accept multiple dependency arguments too.